packages/bierzo-wallet/src/store/tokens/actions.ts

```diff
@@ -1,18 +1,45 @@
+import { BlockchainConnection } from '@iov/bcp';
+import { BnsConnection } from '@iov/bns';
 import { EthereumConnection } from '@iov/ethereum';
+import { LiskConnection } from '@iov/lisk';
+import { ChainSpec, loadConfig } from '../../utils/config';
 import { AddTickerActionType, BwToken } from './reducer';
 
-export async function getTokens(): Promise<{ [key: string]: BwToken }> {
+async function createConnectionFor(spec: ChainSpec): Promise<BlockchainConnection | null> {
+  const url = spec.bootstrapNodes[0];
+  if (spec.codecType === 'eth') {
+    return EthereumConnection.establish(url, {});
+  }
+
+  if (spec.codecType === 'bns') {
+    return BnsConnection.establish(url);
+  }
+
+  if (spec.codecType === 'lsk') {
+    return LiskConnection.establish(url);
+  }
+
+  return null;
```

Comment:
This case indicates a programming error (got `spec` we do not support), so throwing an exception is better than returning null because the caller code does not need to deal with this case.